motoko/passwords/src/passwords/main.mo

```diff
@@ -0,0 +1,121 @@
+import Array "mo:base/Array";
+import Principal "mo:base/Principal";
+import Option "mo:base/Option";
+
+actor {
+
+  // Types
+
+  type Credentials = {
+    username : Text;
+    password : Text;
+  };
+
+  type Account = {
+    id : Nat;
+    username : Text;
+    hash : Text;
+    salt : Text;
+  };
+
+  type Session = {
+    principal : Principal;
+    account_id : Nat;
+  };
+
+
+  // Data Structures
+
+  // Arrays a sub-optimal choice here but they're easy to understand
+  var accounts : [Account] = [];
+  var sessions : [Session] = [];
+  var next_account_id : Nat = 0;
+
+  // Public Functions
+
+  public func signup (credentials : Credentials) {
+    // Generate a random salt and hash it together with the password
+    let _salt = random_salt();
+    let _hash = sha256(credentials.password # _salt);
```

Comment:
`bcrypt` (et al) may get less complaints than `sha256`. My understanding is that `sha256` is too fast to calculate, so too succeptible to brute-force attacks (if you can access the 'secret' password hash, then just guess all the passwords and hash each until you find a hash that matches)

https://cwe.mitre.org/data/definitions/916.html
https://dusted.codes/sha-256-is-not-a-secure-password-hashing-algorithm

Reply:
I like how you called `random_salt()` a good name even with a stubbed implementation.
Could be illustrative to also have a `hashPassword(password, _salt)` function (that for now is just stubbed to call out to 'bad' `sha256). Purely editorial decision by you tho.